res/qml/WaveformDisplay.qml

```diff
@@ -0,0 +1,247 @@
+import "." as Skin
+import Mixxx 1.0 as Mixxx
+import Mixxx.Controls 1.0 as MixxxControls
+import QtQuick 2.12
+import "Theme"
+
+Item {
+    id: root
+
+    required property string group
+
+    enum MouseStatus {
+        Normal,
+        Bending,
+        Scratching
+    }
+
+    MixxxControls.WaveformDisplay {
+        anchors.fill: parent
+        group: root.group
+        zoom: zoomControl.value
+        backgroundColor: "#5e000000"
+
+        Mixxx.WaveformRendererEndOfTrack {
+            color: '#ff8872'
+            endOfTrackWarningTime: 30
+        }
+
+        Mixxx.WaveformRendererPreroll {
+            color: '#ff8872'
+        }
+
+        Mixxx.WaveformRendererMarkRange {
+            // <!-- Loop -->
+            Mixxx.WaveformMarkRange {
+                startControl: "loop_start_position"
+                endControl: "loop_end_position"
+                enabledControl: "loop_enabled"
+                color: '#00b400'
+                opacity: 0.7
+                disabledColor: '#FFFFFF'
+                disabledOpacity: 0.6
+            }
+            // <!-- Intro -->
+            Mixxx.WaveformMarkRange {
+                startControl: "intro_start_position"
+                endControl: "intro_end_position"
+                color: '#2c5c9a'
+                opacity: 0.6
+                durationTextColor: '#ffffff'
+                durationTextLocation: 'after'
+            }
+            // <!-- Outro -->
+            Mixxx.WaveformMarkRange {
+                startControl: "outro_start_position"
+                endControl: "outro_end_position"
+                color: '#2c5c9a'
+                opacity: 0.6
+                durationTextColor: '#ffffff'
+                durationTextLocation: 'before'
+            }
+        }
+
+        Mixxx.WaveformRendererRGB {
+            axesColor: '#a1a1a1a1'
+            lowColor: '#ff2154d7'
+            midColor: '#cfb26606'
+            highColor: '#e5029c5c'
+
+            gainAll: 1.0
+            gainLow: 1.0
+            gainMid: 1.0
+            gainHigh: 1.0
+        }
+
+        Mixxx.WaveformRendererStem {
+            gainAll: 1.0
+        }
+
+        Mixxx.WaveformRendererBeat {
+            color: '#a1a1a1a1'
+        }
+
+        Mixxx.WaveformRendererMark {
+            playMarkerColor: 'cyan'
+            playMarkerBackground: 'orange'
+            defaultMark: Mixxx.WaveformMark {
+                align: "bottom|right"
+                color: "#00d9ff"
+                textColor: "#1a1a1a"
+                text: " %1 "
+            }
+
+            untilMark.showTime: true
+            untilMark.showBeats: true
+            untilMark.align: Mixxx.WaveformUntilMark.AlignBottom
+            untilMark.textSize: 11
+
+            Mixxx.WaveformMark {
```

Comment:
Btw, in the long term I'd like to be able to define a mark delegate from QML. That would allow different skins to have a way to define a custom look (not just custom colors) for waveform marks.

Maybe by defining a QML item here, which is then rendered to a pixmap and then copied to the waveform. Nothing for this PR, but this would be a long term goal.